development/docs/build_block_docs.py

```diff
@@ -0,0 +1,444 @@
+        result = ", ".join(set(result))
+        return f"Union[{result}]"
+    return result[0]
+
+
+def format_inputs(block_definition: dict) -> List[Tuple[str, str, str, bool]]:
+    global_result = []
+    properties = block_definition['properties']
+    for property_name, property_definition in properties.items():
+        if property_name == 'type':
+            continue
+        if property_definition.get("type") in TYPE_MAPPING:
+            result = TYPE_MAPPING[property_definition["type"]]
+            global_result.append((property_name, result, property_definition.get("description", "not available"), False))
+            continue
+        if 'items' in property_definition:
+            if "reference" in property_definition["items"]:
+                continue
+            t_name, ref_appears = create_array_typing(property_definition["items"])
+            global_result.append((property_name, t_name, property_definition.get("description", "not available"), ref_appears))
+            continue
+        if 'anyOf' in property_definition or 'oneOf' in property_definition or 'allOf' in property_definition:
+            x = property_definition.get('anyOf', []) + \
+                property_definition.get('oneOf', []) + \
+                property_definition.get('allOf', [])
+            primitive_types = [e for e in x if "reference" not in e]
+            if len(primitive_types) == 0:
+                continue
+            ref_appears = len(primitive_types) != len(x)
+            result = []
+            for t in primitive_types:
+                if "$ref" in t:
+                    t_name = t["$ref"].split("/")[-1]
+                elif t["type"] in TYPE_MAPPING:
+                    t_name = TYPE_MAPPING[t["type"]]
+                elif t["type"] == "array":
+                    t_name, ref_appears_nested = create_array_typing(t)
+                    ref_appears = ref_appears or ref_appears_nested
+                else:
+                    t_name = "unknown"
+                result.append(t_name)
+            result = set(result)
+            if "None" in result:
+                high_level_type = "Optional"
+                result.remove("None")
+            else:
+                high_level_type = "Union"
+            result_str = ", ".join(list(result))
+            if len(primitive_types) > 1:
+                result_str = f"{high_level_type}[{result_str}]"
+            global_result.append((property_name, result_str, property_definition.get("description", "not available"), ref_appears))
+        if 'reference' in property_definition:
+            continue
+    return global_result
+
+
+def create_array_typing(array_definition: dict) -> Tuple[str, bool]:
+    ref_appears = False
+    high_level_type = "Set" if array_definition.get("uniqueItems", False) is True else "List"
+    if len(array_definition.get("items", [])) == 0:
+        return f"{high_level_type}[Any]", ref_appears
+    if "type" in array_definition["items"]:
```

Comment:
I would change into `if "type" not in ..." -> return 
and then resign from one level of indent